### ISSUE TITLE
Tick data with millisecond component doesn't reach consolidator

### ISSUE THREAD

**Issue by jaredbroad** [QuantConnect/Lean#796]
User reported issue with the Algorithm Manager update consolidator logic:
https://github.com/QuantConnect/Lean/blob/master/Engine/AlgorithmManager.cs#L480

Tick resolution data is set to check for TimeSpan.FromSeconds(0) but the minor variation in the sampling time of the ticks can mean they have tiny millisecond components which will make the filter not match for grouped ticks.

User suggested fix changing:
```
var algorithmTimeSpan = resolutionTimeSpan == TimeSpan.FromTicks(0)
                                   ? TimeSpan.FromTicks(0)
                                   : TimeSpan.FromSeconds(1);
to:
var algorithmTimeSpan = resolutionTimeSpan == TimeSpan.FromTicks(0)
                                   ? TimeSpan.FromMilliseconds(30)
                                   : TimeSpan.FromSeconds(1);
```
This logic was designed to filter out fillforward bars, so only real bars update the indicators. E.g. Imagine we have a system where you can have multiple resolutions of data in your algorithm:
SPY - Hourly
IBM - Minute
But each OnData will have both SPY and IBM; as the SPY bar is copied forward for every minute.

The update filter logic makes sure we only add the bar to consolidator which matches the consolidator's resolution. This is when it rounds down to the same value as the config datafeed itself.

**Comment by blaursen** [comment]
The fix changing is a very temporary solution and let me add the following:

The fix change
```
var algorithmTimeSpan = resolutionTimeSpan == TimeSpan.FromTicks(0)
                                   ? TimeSpan.FromMilliseconds(30)
                                   : TimeSpan.FromSeconds(1);
```

Will only work if the logic
https://github.com/QuantConnect/Lean/blob/master/Engine/AlgorithmManager.cs#L483
is changed:
from:
```
if (algorithm.UtcTime.RoundDown(algorithmTimeSpan) == dataPoint.EndTime.RoundUp(resolutionTimeSpan).ConvertToUtc(update.Target.ExchangeTimeZone))
{
                                    consolidator.Update(dataPoint);
}
```
to: (using less than instead of equality)
```
if (algorithm.UtcTime.RoundDown(algorithmTimeSpan) <= dataPoint.EndTime.RoundUp(resolutionTimeSpan).ConvertToUtc(update.Target.ExchangeTimeZone))
 {
                                    consolidator.Update(dataPoint);
 }
```

This will introduce the risc of throwing a `ArgumentException` in
https://github.com/QuantConnect/Lean/blob/master/Indicators/IndicatorBase.cs#L80

Also, let me just point out a concrete example of `RoundDown`, `RoundUp` which  shows that the greater than inequality above is necessary. We use the rounding TimeSpan `TimeSpan.FromMilliSeconds(30)`

Now: 16:08:55.6760+01:00
RounddownNow: 16:08:55.6500+01:00
RoundupNow: 16:08:55.6800+01:00

Which was generated from the code:
```
DateTime now = DateTime.Now;  
var rounding = TimeSpan.FromMilliseconds(30);
var nowRoundDown = now.RoundDown(rounding);
var nowRoundUp = now.RoundUp(rounding);
string format = "HH:mm:ss.ffffzzz";
Log(string.Format("\nNow: {0}\nRounddownNow: {1}\nRoundupNow: {2}", now.ToString(format), nowRoundDown.ToString(format), nowRoundUp.ToString(format)));
```

**Comment by jaredbroad** [comment]
Closed by #814
